js/ui/metric-popup.js

```diff
@@ -62,9 +39,175 @@ Vue.component('metric-popup', {
+                }
+              }
+            }
+            if (paramMyOldMetric.color !== paramMyMetric.color) {
+              paramMyMetric.updateColor(paramMyMetric.color)
+              const colorEle = document.getElementsByClassName(paramMyMetric.popupKey)
+              if (colorEle && colorEle[0]) {
+                colorEle[0].style.color = paramMyMetric.color
+              }
+            }
+            if (nameChanged) {
+              // TODO: do something, in this case, do forceUpdate the legendApp
+              //         so the metric's color will be shown
+            }
+            if (paramMyOldMetric.marker !== paramMyMetric.marker) {
+              paramMyMetric.updateMarker(paramMyMetric.marker)
+            }
+            // don't do a complete repaint
+            // renderMetrics();
+          }
+        }
+      }
+    }(this.metric, this.oldMetric, instance, affectedTraces))
+    veil.create(disablePopupFunc)
+    veil.attachPopup(popupEle)
+
+    const colorchooserEle = popupEle.querySelector('.popup_colorchooser')
+    const colorchooserObj = new Colorchooser(colorchooserEle, this.metric)
+    colorchooserObj.onchange = (function (myTraces, paramMyMetric) {
+      return function () {
+        paramMyMetric.renderer.graticule.draw(false)
+      }
+    }(affectedTraces, this.metric))
+
+    document.getElementById('input_metric_name').focus()
+  },
   methods: {
-    changeMarker: function () {
-      this.metric.updateMarker(document.querySelector('.popup_legend_select').value)
+    changeMarker: function (evt) {
+      this.metric.updateMarker(evt.target.value)
+    },
+    trashcanClicked: function (evt) {
+      veil.destroy(evt)
+    },
+    metricNameKeyup: function (evt) {
+      const instance = window.MetricQWebView.instances[0]
+      if (evt.key.toLowerCase() === 'enter') {
+        veil.destroy(evt)
+      }
+      // TODO: implement throttling?
+      instance.handler.searchMetricsPromise(evt.target.value).then(
+        function (searchSuggestions) {
+          const datalistEle = document.getElementById('autocomplete_metric')
+          if (!datalistEle) {
+            showUserHint('Auto-Vervollständigung nicht verfügbar, konnte Element #autocomplete_metric nicht finden.')
+          } else {
+            for (let i = datalistEle.childNodes.length - 1; i >= 0; --i) {
+              datalistEle.removeChild(datalistEle.childNodes[i])
+            }
+            for (let i = 0; i < searchSuggestions.length; ++i) {
+              const optionEle = document.createElement('option')
+              optionEle.setAttribute('value', searchSuggestions[i])
+              datalistEle.appendChild(optionEle)
+            }
+          }
+        }
+      )
+    },
+    closePopup: function (evt) {
+      veil.destroy(evt)
+    },
+    closePopupModal: function (evt) {
+      if (evt.target.getAttribute('role') === 'dialog') {
+        veil.destroy(evt)
+      }
```

Comment:
I'm confused by the popup handling.
There's the state in Store.
There's are different close functions, but they don't use the the store.
Then there toggle event.
This looks like plenty of things that can and will go wrong.

Reply:
`veil` is unfortunately not only UI-sugar, but also handles popup/veil destruction and sometimes saving. I think this is not the scope of this PR and should be addressed in a new one (see #47)